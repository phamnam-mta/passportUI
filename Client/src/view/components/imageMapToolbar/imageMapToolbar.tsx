import * as React from "react";
import { DefaultIconButton } from "view/components/buttons/buttons";

interface IImageMapToolbarProps {
    disabled: boolean;
    zoomRatio?: number;
    rotateAngle: number;
    onZoomInClick: () => void;
    onZoomOutClick: () => void;
    onZoomToFitClick: () => void;
    onRotateClick: () => void;
}

export class ImageMapToolbar extends React.PureComponent<IImageMapToolbarProps> {
    public render() {
        const { disabled, onZoomInClick, onZoomOutClick, onZoomToFitClick, onRotateClick, zoomRatio, rotateAngle } =
            this.props;

        const ratio = zoomRatio ? parseInt(String(zoomRatio * 100)) + "%" : "";
        const angle = (rotateAngle % 360) + "Â°";
        return (
            <div className="image-map-tool-bar">
                <DefaultIconButton
                    name="ZoomIn"
                    title="Zoom in"
                    ariaLabel={ratio}
                    onClick={onZoomInClick}
                    disabled={disabled}
                />
                <DefaultIconButton
                    name="ZoomOut"
                    title="Zoom out"
                    ariaLabel={ratio}
                    onClick={onZoomOutClick}
                    disabled={disabled}
                />
                <DefaultIconButton
                    name="ZoomToFit"
                    title="Zoom to fit"
                    onClick={onZoomToFitClick}
                    disabled={disabled}
                />
                <DefaultIconButton
                    name="Rotate"
                    title="Rotate"
                    ariaLabel={angle}
                    onClick={onRotateClick}
                    disabled={disabled}
                />
            </div>
        );
    }
}

export default ImageMapToolbar;
