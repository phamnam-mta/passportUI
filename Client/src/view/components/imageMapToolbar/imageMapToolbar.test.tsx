import { shallow } from "enzyme";
import * as React from "react";
import { ImageMapToolbar } from "./imageMapToolbar";

describe("<ImageMapToolbar />", () => {
    let baseProps;

    beforeEach(() => {
        baseProps = {
            disabled: false,
            zoomRatio: 1.2,
            rotateAngle: 0,
            onZoomInClick: jest.fn(),
            onZoomOutClick: jest.fn(),
            onZoomToFitClick: jest.fn(),
            onRotateClick: jest.fn(),
        };
    });

    describe("Rendering", () => {
        it("should render correctly", () => {
            const wrapper = shallow(<ImageMapToolbar {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it is disabled", () => {
            const props = {
                ...baseProps,
                disabled: true,
            };

            const wrapper = shallow(<ImageMapToolbar {...props} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Event Handling", () => {
        const buttonSelector = "DefaultIconButton";

        it("should call zoom-in handler, when ZoomIn button is clicked", () => {
            const wrapper = shallow(<ImageMapToolbar {...baseProps} />);
            const button = wrapper.find(buttonSelector).first();
            button.simulate("click");

            expect(baseProps.onZoomInClick).toBeCalledTimes(1);
        });

        it("should call zoom-out handler, when ZoomOut button is clicked", () => {
            const wrapper = shallow(<ImageMapToolbar {...baseProps} />);
            const button = wrapper.find(buttonSelector).at(1);
            button.simulate("click");

            expect(baseProps.onZoomOutClick).toBeCalledTimes(1);
        });

        it("should call zoom-to-fit handler, when ZoomToFit button is clicked", () => {
            const wrapper = shallow(<ImageMapToolbar {...baseProps} />);
            const button = wrapper.find(buttonSelector).at(2);
            button.simulate("click");

            expect(baseProps.onZoomToFitClick).toBeCalledTimes(1);
        });

        it("should call rotate handler, when Rotate button is clicked", () => {
            const wrapper = shallow(<ImageMapToolbar {...baseProps} />);
            const button = wrapper.find(buttonSelector).last();
            button.simulate("click");

            expect(baseProps.onRotateClick).toBeCalledTimes(1);
        });
    });
});
