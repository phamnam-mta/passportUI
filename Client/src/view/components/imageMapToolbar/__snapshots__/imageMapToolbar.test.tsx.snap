// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageMapToolbar /> Rendering should match snapshot, when it is disabled 1`] = `
<div
  className="image-map-tool-bar"
>
  <DefaultIconButton
    ariaLabel="120%"
    disabled={true}
    name="ZoomIn"
    onClick={[MockFunction]}
    title="Zoom in"
  />
  <DefaultIconButton
    ariaLabel="120%"
    disabled={true}
    name="ZoomOut"
    onClick={[MockFunction]}
    title="Zoom out"
  />
  <DefaultIconButton
    disabled={true}
    name="ZoomToFit"
    onClick={[MockFunction]}
    title="Zoom to fit"
  />
  <DefaultIconButton
    ariaLabel="0°"
    disabled={true}
    name="Rotate"
    onClick={[MockFunction]}
    title="Rotate"
  />
</div>
`;

exports[`<ImageMapToolbar /> Rendering should render correctly 1`] = `
<div
  className="image-map-tool-bar"
>
  <DefaultIconButton
    ariaLabel="120%"
    disabled={false}
    name="ZoomIn"
    onClick={[MockFunction]}
    title="Zoom in"
  />
  <DefaultIconButton
    ariaLabel="120%"
    disabled={false}
    name="ZoomOut"
    onClick={[MockFunction]}
    title="Zoom out"
  />
  <DefaultIconButton
    disabled={false}
    name="ZoomToFit"
    onClick={[MockFunction]}
    title="Zoom to fit"
  />
  <DefaultIconButton
    ariaLabel="0°"
    disabled={false}
    name="Rotate"
    onClick={[MockFunction]}
    title="Rotate"
  />
</div>
`;
